pkg/store/kotsstore/downstream_store.go

```diff
@@ -409,7 +409,20 @@ func (s *KOTSStore) GetDownstreamVersions(appID string, clusterID string, downlo
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to get app license")
 	}
-	downstreamtypes.SortDownstreamVersions(result.AllVersions, license.Spec.IsSemverRequired)
+
+	foundChannelID, err := s.GetAppChannelID(appID)
+	var licenseChan *kotsv1beta1.Channel
+	if foundChannelID == "" {
```

Comment:
i see this code duplicated in numerous locations. can it be moved to a shared function? something like `BackfillAppChannelID`.